tensorflow/python/ops/math_ops.py

```diff
@@ -4857,7 +4858,207 @@ def sparse_segment_sum(
+  array([[[0, 1],
+          [1, 0]],
+         [[0, 0],
+          [1, 0]]], dtype=int32)>
+  >>> values = tf.constant([3, 5, 2, 7], shape=[2, 2])
+  >>> values
+  <tf.Tensor: shape=(2, 2), dtype=int32, numpy=
+  array([[3, 5],
+         [2, 7]], dtype=int32)>
+  >>> dense_shape = tf.constant([2, 2])
+  >>> dense_shape
+  <tf.Tensor: shape=(2,), dtype=int32, numpy=array([2, 2], dtype=int32)>
+  >>> mat1 = tf.constant(np.arange(1, 13), shape=[2, 2, 3])
+  >>> mat1
+  <tf.Tensor: shape=(2, 2, 3), dtype=int64, numpy=
+  array([[[ 1,  2,  3],
+          [ 4,  5,  6]],
+         [[ 7,  8,  9],
+          [10, 11, 12]]])>
+  >>> mat2 = tf.constant(np.arange(13, 25), shape=[2, 3, 2])
+  >>> mat2
+  <tf.Tensor: shape=(2, 3, 2), dtype=int64, numpy=
+  array([[[13, 14],
+          [15, 16],
+          [17, 18]],
+         [[19, 20],
+          [21, 22],
+          [23, 24]]])>
+  >>> tf.sparse.sampled_addmm(indices, values, dense_shape, mat1, mat2,
+  ... alpha=0.75, beta=0.25)
+  (<tf.Tensor: shape=(2, 2, 2), dtype=int32, numpy=
+  array([[[0, 1],
+          [1, 0]],
+         [[0, 0],
+          [1, 0]]], dtype=int32)>, <tf.Tensor: shape=(2, 2), dtype=float32,
+  numpy=array([[ 75.75, 173.  ],
+         [381.5 , 524.5 ]], dtype=float32)>, 
+  <tf.Tensor: shape=(2,), dtype=int32, numpy=array([2, 2], dtype=int32)>)
 
+  Args:
+    indices: `tf.Tensor` containing coordinates for the rows and columns to be
+      multiplied. Must have rank > 1.
+    values: `tf.Tensor` containing the values to be scaled and added to the
+      sampled dot product.
+    dense_shape: `tf.Tensor` defining the dense shape of the output.
+    mat1: `tf.Tensor` to be multiplied. Must have rank > 1.
+    mat2: `tf.Tensor` to be multiplied. Must have rank > 1.
+    beta: Number to be multipled with `values`. Defaults to 1.0.
+    alpha: Number to be multiplied with the sampled dot product of `mat1` and
+      `mat2`. Defaults to 1.0.
+    output_type: the output datatype if needed. Defaults to float32.
+
+  Returns:
+    A tuple of three `tf.Tensor` objects (indices, result, dense_shape) making
+    up the components of a `SparseTensor` representing the result of the
+    operation.
+
+    result = `alpha` * (`mat1` @ `mat2`) * spy(`indices`) + `beta` * `values`
+
+    The function `spy(indices)` is the sparsity pattern matrix derived from
+    `indices`. 
+  """
+  indices = ops.convert_to_tensor(indices)
+  values = ops.convert_to_tensor(values, dtype=output_type)
+  dense_shape = ops.convert_to_tensor(dense_shape, dtype=dtypes.int32)
+  mat1 = ops.convert_to_tensor(mat1, dtype=output_type)
+  mat2 = ops.convert_to_tensor(mat2, dtype=output_type)
+
+  mat1_shape = mat1.shape
+  mat2_shape = mat2.shape
+  
+  if mat1_shape is None:
+    mat1_shape = array_ops.shape(mat1)
+  if mat2_shape is None:
+    mat2_shape = array_ops.shape(mat2)
+
+  dense_rows = mat1_shape[-2]
+  dense_cols = mat2_shape[-1]
+
+  output_shape = constant_op.constant([dense_rows, dense_cols], 
```

Comment:
I don't think this will work if the inputs are not constant.  You'll need to use something like `array_ops.concat()`.

Reply:
`array_ops.concat()` wouldn't let me pass in two scalars so I tried `array_ops_stack.stack()` instead